### ISSUE TITLE
Unit equivalences in operations and decompose

### ISSUE THREAD

**Issue by Cadair** [astropy/astropy#3990]
@Alex-Ian-Hamilton and I have been playing around  with custom equivalences and we ran into an interesting issue.

Normally if you divide one unit by another equivalent unit you get:

```python
In [1]: import astropy.units as u

In [2]: a = 10*u.m

In [3]: b = 10*u.cm

In [4]: a/b
Out[4]: <Quantity 1.0 m / cm>

In [5]: (a/b).decompose()
Out[5]: <Quantity 100.0>
```
Which makes sense, it's a dimensionless unit because you divided two units of the same type by each other.

What gets interesting is if you try this with an explicit equivalencey:

```python
In [6]: u.set_enabled_equivalencies(u.spectral())
Out[6]: <astropy.units.core._UnitContext at 0x7f7cc8dfcad0>

In [8]: b = 10*u.Hz

In [9]: a/b
Out[9]: <Quantity 1.0 m / Hz>

In [10]: (a/b).decompose()
Out[10]: <Quantity 1.0 m s>

In [11]: (a/b).to(u.one)
---------------------------------------------------------------------------
UnitsError                                Traceback (most recent call last)
<ipython-input-11-70e9e9e387aa> in <module>()
----> 1 (a/b).to(u.one)
/home/stuart/GitHub/astropy/astropy/units/quantity.pyc in to(self, unit, equivalencies)
    604         unit = Unit(unit)
    605         new_val = np.asarray(
--> 606             self.unit.to(unit, self.value, equivalencies=equivalencies))
    607         return self._new_view(new_val, unit)
    608 
/home/stuart/GitHub/astropy/astropy/units/core.pyc in to(self, other, value, equivalencies)
    945             If units are inconsistent
    946         """
--> 947         return self._get_converter(other, equivalencies=equivalencies)(value)
    948 
    949     def in_units(self, other, value=1.0, equivalencies=[]):

/home/stuart/GitHub/astropy/astropy/units/core.pyc in _get_converter(self, other, equivalencies)
    849         except UnitsError:
    850             return self._apply_equivalencies(
--> 851                 self, other, self._normalize_equivalencies(equivalencies))
    852         return lambda val: scale * _condition_arg(val)
    853 

/home/stuart/GitHub/astropy/astropy/units/core.pyc in _apply_equivalencies(self, unit, other, equivalencies)
    840         raise UnitsError(
    841             "{0} and {1} are not convertible".format(
--> 842                 unit_str, other_str))
    843 
    844     def _get_converter(self, other, equivalencies=[]):

UnitsError: 'm / Hz' and '' (dimensionless) are not convertible

```

That doesn't seem right to my mind, I have just told units that `Hz` and `m` are equivalent, so I would expect similar behaviour to them being of the same base type.

Basically I think that it should be possible to specify equivalences to `decompose()`.

**Comment by Cadair** [comment]
ping @astrofrog

**Comment by mdboom** [comment]
I see where you're going.  The difficulty with supporting equivalencies in decompose is that without a destination (as you have in `to`) you can't know when to apply the equivalencies.  Is it to maximize some measure of "simplicity" of the expression (which is hard to define, in experience -- is it lowest powers, fewest number of base units...?)

**Comment by mhvk** [comment]
I'm looking over old unit issues and closing this as I don't see an easy way forward at all. I don't think an equivalency can apply to a composite unit.
